includes/class-wc-payments-apple-pay-registration.php

```diff
@@ -0,0 +1,357 @@
+		$redirect = 'index.php?' . self::DOMAIN_ASSOCIATION_FILE_NAME . '=1';
+
+		add_rewrite_rule( $regex, $redirect, 'top' );
+	}
+
+	/**
+	 * Add to the list of publicly allowed query variables.
+	 *
+	 * @param  array $query_vars - provided public query vars.
+	 * @return array Updated public query vars.
+	 */
+	public function whitelist_domain_association_query_param( $query_vars ) {
+		$query_vars[] = self::DOMAIN_ASSOCIATION_FILE_NAME;
+		return $query_vars;
+	}
+
+	/**
+	 * Serve domain association file when proper query param is provided.
+	 *
+	 * @param object $wp WordPress environment object.
+	 */
+	public function parse_domain_association_request( $wp ) {
+		if (
+			! isset( $wp->query_vars[ self::DOMAIN_ASSOCIATION_FILE_NAME ] ) ||
+			'1' !== $wp->query_vars[ self::DOMAIN_ASSOCIATION_FILE_NAME ]
+		) {
+			return;
+		}
+
+		$path = WCPAY_ABSPATH . '/' . self::DOMAIN_ASSOCIATION_FILE_NAME;
+		header( 'Content-Type: text/plain;charset=utf-8' );
+		echo esc_html( @file_get_contents( $path ) ); // @codingStandardsIgnoreLine
+		exit;
+	}
+
+	/**
+	 * Processes the Apple Pay domain verification.
+	 *
+	 * @return bool Whether domain verification succeeded.
+	 */
+	public function register_domain_with_apple() {
+		$error = null;
+		try {
+			$registration_response = $this->payments_api_client->register_domain_with_apple( $this->domain_name );
+
+			if ( isset( $registration_response['id'] ) ) {
+				$this->gateway_settings['apple_pay_verified_domain'] = $this->domain_name;
+				$this->gateway_settings['apple_pay_domain_set']      = 'yes';
+				update_option( 'woocommerce_woocommerce_payments_settings', $this->gateway_settings );
+
+				Logger::log( __( 'Your domain has been verified with Apple Pay!', 'woocommerce-payments' ) );
+
+				return true;
+			} elseif ( isset( $registration_response['error']['message'] ) ) {
+				$error = $registration_response['error']['message'];
+			}
+		} catch ( Exception $e ) {
+			$error = $e->getMessage();
+		}
+
+		// Display error message in notice.
+		$this->apple_pay_verify_notice = $error;
+
+		$this->gateway_settings['apple_pay_verified_domain'] = $this->domain_name;
+		$this->gateway_settings['apple_pay_domain_set']      = 'no';
+		update_option( 'woocommerce_woocommerce_payments_settings', $this->gateway_settings );
+
+		Logger::log( 'Error: ' . $error );
+
+		return false;
+	}
+
+	/**
+	 * Process the Apple Pay domain verification if proper settings are configured.
+	 */
+	public function verify_domain_if_configured() {
+		if ( ! $this->is_enabled() ) {
+			return;
+		}
+
+		// Ensure that domain association file will be served.
+		flush_rewrite_rules();
+
+		// The rewrite rule method doesn't work if permalinks are set to Plain.
+		// Create/update domain association file by copying it from the plugin folder as a fallback.
+		$this->update_domain_association_file();
+
+		// Register the domain with Apple Pay.
+		$verification_complete = $this->register_domain_with_apple();
+	}
```

Comment:
Why are we assigning the result from `register_domain_with_apple()` to a variable here? Are we supposed to be returning it?

Or maybe it's fine to ignore the return value here?

Reply:
That was a mistake. A leftover from Stripe. This variable was used to display an inbox note on Stripe, but we are not doing it here. I'm going to remove it 👍